### ISSUE TITLE
Unit cost & balance

### ISSUE THREAD

**Issue by SoapingMaster** [UnofficialCrusaderPatch/UnofficialCrusaderPatch#937]
Hello everyone,

I've recently downloaded this awesome patch for the good old shc. I usually play multiplayer with friends, so I'm more interested in balance changes. I really do appreciate the crossbow nerf and the spearman buff. However, some of the unit costs are still not reasonable.

Is it yet possible to modify the gold cost of units upon recruiting?
The exact values aren't that important for now, but I really look forward to this, as it could maximize the game experience.

Just as a sketch, here are my ideas. After years of playing this game, I do thnik that:
1) arabian archers should be cheaper around 60-70 gold
2) assassins are way too cheap, their low visibility should be an expensive trait for like 80-100 gold
3) horse archers are well known op troops, nothing less than 150 gold would fix them
4) fire ballistae and catapults may be more expensive too
5) as others mentioned, shields should be nerfed too
6) crossbows should be 30 gold

This is only my opinion, and may be flawed. What do you think of this?

**Comment by GRhin** [comment]
First off, we are working on adding the ability for users to set their own balance changes so when that's added you can do whatever you like.
As to your suggestions, based off a rebalance a few of us are working on for multiplayer:
1. Arab archers cheaper could work, making euro archers more expensive is another option, we have done a little of both I think. This would be a controversial one, as we commonly get a request to nerf Arab archers (the math supports buffing however)
2. Assassins are certainly overpowered, but their invisibility isn't undercoated on it's own. We have made them slower (by removing the rally point running speed), and reducing their melee strength, and it seems to be sufficient. Increasing their cost will limit their utility as spreading slingers or archers around your base becomes more and more cost effective, and once walls go up it takes only a few units at each vulnerable resource to effectively counter them.
3. I agree with the supposition that ha are strong, but in single player this is more due to players amassing far more troops than necessary to attack with. That is a problem with AIs not with ha. In multiplayer the ha are indirectly nerfed by by the shields speed nerf, ha can no longer stay behind shields and arent as strong when enemy can match you for troop count and unit composition.
4. We have made some changes to seige, but again, in single player players will spend thousands upon thousands of gold on a seige army, to fight a few hundred gold worth of defending units and not mind much if they lose it all. So increasing coats will not change that, only make the games longer (or players change speed more frequently). In multiplayer they aren't as op as you think as humans can actively counter them with their own seige, shields and tactics.
5. Shields only need a speed nerf in my opinion, but I think there has been an additional health nerf since I last worked on our balance
6. Crossbows are wildly overrated. If you do the math (which I have), crossbows are only better per gold spent than euro archers when defending against swords and knights and a couple others I forget. For the most part it is cheaper to build an army or archers that deal the same amount of damage, and those few exceptions can be easily countered with slingers recruited from the gold saved by going for archers.
Crossbows strength come from their ability to concentrate more firepower in a smaller area, and using less workers (so  best late game when you have more gold than you can use recruiting cheap units at the slow worker spawn rate)

Again, this is my view based on math looking at current stat values, long experience in single player, and from viewing the results of changes made for multiplayer over the last year or so. We are still working on our work, so it is always possible my views on any of this could change as more evidence is seen. 
Keep in mind also that the multiplayer I prefer starts with no gold, so you need to consider gold efficiency much more than in traditional multiplayer styles.

**Comment by Russianstory** [comment]
@GRhin  ONE MINUTE you say that you have already done something ... then the question is - is there a place where you write that HAS ALREADY BEEN DONE in a new pachter, which will only be released in the future? Where can you read about what you have already done? Very interesting. I am personally looking forward to the next release!

**Comment by GRhin** [comment]
No, this has not yet been implemented into the patcher. I have been working on finding all the data points so that it can be implemented in the future. And have found pretty much everything, and am now just waiting on the devs to finish an overhaul of how the patcher works (will make sense with the next release) and have time to look into implementing the balance updates, which is not a small feature so will take time.
For my testing I have a tool that modifies values at specific exe addresses that was made to help me test the addresses I found, and we use that to test the balance changes as well untill the patcher makes it easy. Forming this rebalance also helps us find places that don't work correctly, for example worker movement speed and several other stats is constantly overwritten so isn't as easy to implement (therefore will not be available in the first version of stats)

TLDR
Rebalance will be in a future ucp version, and hopefully not too far into the future. 
There is not a balance tool around to use now, but a developer utility tool is being used as a very complicated workaround for testing purposes untill a developer can work on making an actual balance tool.
